src/Runner.Common/JobServerQueue.cs

```diff
@@ -459,19 +480,30 @@ private async Task ProcessSummaryUploadQueueAsync(bool runOnce = false)
                 {
                     if (runOnce)
                     {
-                        Trace.Info($"Uploading {filesToUpload.Count} summary files in one shot through results service.");
+                        Trace.Info($"Uploading {filesToUpload.Count} file(s) in one shot through results service.");
                     }
 
                     int errorCount = 0;
                     foreach (var file in filesToUpload)
                     {
                         try
                         {
-                            await UploadSummaryFile(file);
+                            if (String.Equals(file.Type, ChecksAttachmentType.StepSummary, StringComparison.OrdinalIgnoreCase))
+                            {
+                                await UploadSummaryFile(file);
+                            }
+                            else if (String.Equals(file.Type, CoreAttachmentType.ResultsLog, StringComparison.OrdinalIgnoreCase))
+                            {
+                                if (file.RecordId != _jobTimelineRecordId)
```

Comment:
do we need this filter? i thought you already filter it out in the enqueue part.

Reply:
Well, this is written in mind for the next PR.  I am removing the filter up there already.